dlt/common/schema/schema.py

```diff
@@ -206,6 +207,16 @@ def update_schema(self, partial_table: TPartialTableSchema) -> TPartialTableSche
             partial_table = utils.merge_tables(table, partial_table)
         return partial_table
 
+
+    def update_schema(self, schema: "Schema") -> None:
+        """Updates this schema from an incoming schema"""
+        # update all tables
+        for table in schema.tables.values():
+            self.update_table(table)
+        # update other settings
```

Comment:
it does not seem we have to update the _settings here, because they are not set within the source, but I am not 100% sure. Lmk what else we need to take care here..

Reply:
I'd just overwrite the settings from incoming schema.